sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -987,6 +1051,64 @@ public virtual Response<AlertingHook> CreateHook(AlertingHook hook, Cancellation
             }
         }
 
+        /// <summary>
+        /// </summary>
+        public virtual async Task<Response> UpdateHookAsync(string hookId, AlertingHook hook, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(hook, nameof(hook));
+
+            Guid hookGuid = ClientCommon.ValidateGuid(hookId, nameof(hookId));
+
+            if (hook is EmailHook emailHook)
+            {
+                Argument.AssertNotNullOrEmpty(emailHook.EmailsToAlert, nameof(EmailHook.EmailsToAlert));
+            }
```

Comment:
It's a bit weird to check this in an Update method, but I agree it's necessary for now. Users are forced to set `EmailsToAlert` in the constructor anyway, so we'll send it, even though it's not required by the service. 😞